draft-ietf-lamps-dilithium-certificates.md

```diff
@@ -997,60 +999,80 @@ The following is the third example:
 ~~~
 
 
-# Pre-hashing (ExternalMu-ML-DSA) {#prehash}
+# Pre-hashing via External Mu {#externalmu}
 
 Some applications require pre-hashing, where the signature generation
-process can be separated into a pre-hash step and a core signature
-step in order to ease operational requirements around large or
-inconsistently-sized payloads. Pre-hashing can be performed at the
-protocol layer, but not all protocols support it. Examples in
-{{RFC5280}} are certificates and CRLs; these do not include message
-digesting before signing. This can make signing large CRLs or a high
-volume of certificates with large public keys challenging.
-
-As mentioned in the introduction, pure ML-DSA signing itself
-supports a pre-hashing flow by splitting the operation over two
-modules. In this section, we make this "ExternalMu-ML-DSA"
-more explicit.
-
-There are two steps. First an `ExternalMu-ML-DSA.Prehash()`
-followed by `ExternalMu-ML-DSA.Sign()`. Together these are functionally
-equivalent to `ML-DSA.Sign()` from {{FIPS204}} in that used in sequence
-they create exactly the same signatures as regular pure ML-DSA, which
-can be verified by the unmodified `ML-DSA.Verify()`.
-
-An ML-DSA key and certificate can be used with either ML-DSA
-or ExternalMu-ML-DSA interchangeably.
-Note that ExternalMu-ML-DSA describes a different signature API from ML-DSA
-and therefore might require explicit support from hardware or
-software cryptographic modules.
-
-Note that the signing mode defined here is different from HashML-DSA
-defined in Section 5.4 of {{FIPS204}}. This specification uses exclusively
-ExternalMu-ML-DSA for pre-hashed use cases. See {{sec-disallow-hash}} for
-additional discussion of why HashML-DSA is disallowed in PKIX.
+process can be separated into a pre-hash step requiring only the message,
+context value, and other public information, and a core signature
+step requiring the secret key in order to ease operational requirements around large or
+inconsistently-sized payloads.
+
+Prehashing can be performed with the HashML-DSA algorithm defined in Section 5.4 of {{FIPS204}}
+however the HashML-DSA algorithm comes with a few disadvantages.
+First, there is a small performance disadvantage in that the pre-hashing step
+needs to be done in addition to the message digesting internal to `ML-DSA.Sign()`.
+Second and more importantly, ML-DSA and HashML-DSA require different `Verify()` routines.
+To address this problem, the NIST Computer Security Objects Register (CSOR)
+has assigned different Object Identifiers (OIDs) to ML-DSA and HashML-DSA.
+This can pose problematic for certain deployments, for example if the
+ML-DSA public is bound to either the ML-DSA or HashML-DSA OID at key generation
+time, for example in an X.509 certificate, then this prevents that key
+from being able to sometimes prehash and sometimes not depending on application need.
```

Comment:
"from being able to sometimes prehash and sometimes not depending on application need" feels a little informal.